homeassistant/components/tuya/camera.py

```diff
@@ -81,6 +89,23 @@ async def stream_source(self) -> str | None:
             "rtsp",
         )
 
+    async def async_create_stream(self) -> Stream | None:
+        """Create a Stream for stream_source."""
+        if not self.stream:
+            return await super().async_create_stream()
+
+        # Tuya streams urls are one time urls. Updating source of existing stream.
+        if not self._create_stream_lock:
+            self._create_stream_lock = asyncio.Lock()
+        async with self._create_stream_lock:
+            async with async_timeout.timeout(CAMERA_STREAM_SOURCE_TIMEOUT):
```

Comment:
This seems like it applies to any call to `stream_source` so want to move it there?

Reply:
I want not to refactor the camera base class.